homeassistant/components/hardware/models.py

```diff
@@ -0,0 +1,23 @@
+"""Models for Hardware."""
+from __future__ import annotations
+
+from typing import Protocol, TypedDict
+
+from homeassistant.core import HomeAssistant, callback
+
+
+class HardwareInfo(TypedDict):
```

Comment:
Maybe make it a dataclass instead? That's more friendly to use. We can easily turn that into a dict as needed later.